common-go-libs/apis/dp/v1alpha1/authentication_conversion.go

```diff
@@ -31,65 +31,72 @@ func (src *Authentication) ConvertTo(dstRaw conversion.Hub) error {
-		Header:              src.Spec.Default.AuthTypes.Oauth2.Header,
-		SendTokenToUpstream: src.Spec.Default.AuthTypes.Oauth2.SendTokenToUpstream,
-	}
-
-	// Convert Oauth2Auth override to v1alpha2.Oauth2Auth : Required field added as mandatory for OAuth2
-	dst.Spec.Override.AuthTypes.Oauth2 = v1alpha2.Oauth2Auth{
-		Required:            "mandatory",
-		Disabled:            src.Spec.Default.AuthTypes.Oauth2.Disabled,
-		Header:              src.Spec.Default.AuthTypes.Oauth2.Header,
-		SendTokenToUpstream: src.Spec.Default.AuthTypes.Oauth2.SendTokenToUpstream,
-	}
-
-	// Convert Oauth2Auth Default to v1alpha2.APIKey : Required field added as optional for APIKey
-	for _, apiKeyAuth := range src.Spec.Default.AuthTypes.APIKey {
-		convertedAPIKeyAuth := v1alpha2.APIKeyAuth{
-			In:                  apiKeyAuth.In,
-			Name:                apiKeyAuth.Name,
-			SendTokenToUpstream: apiKeyAuth.SendTokenToUpstream,
-		}
-		dst.Spec.Default.AuthTypes.APIKey = append(dst.Spec.Default.AuthTypes.APIKey, convertedAPIKeyAuth)
-	}
-
-	// Convert Oauth2Auth Override to v1alpha2.APIKey : Required field added as optional for APIKey
-	for _, apiKeyAuth := range src.Spec.Override.AuthTypes.APIKey {
-		convertedAPIKeyAuth := v1alpha2.APIKeyAuth{
-			In:                  apiKeyAuth.In,
-			Name:                apiKeyAuth.Name,
-			SendTokenToUpstream: apiKeyAuth.SendTokenToUpstream,
-		}
-		dst.Spec.Override.AuthTypes.APIKey = append(dst.Spec.Override.AuthTypes.APIKey, convertedAPIKeyAuth)
-	}
-
-	// Convert testConsoleKey Override to v1alpha2.JWT
-	if src.Spec.Override.AuthTypes.TestConsoleKey != (TestConsoleKeyAuth{}) {
-		dst.Spec.Override.AuthTypes.JWT = v1alpha2.JWT{
-			Header:              src.Spec.Override.AuthTypes.TestConsoleKey.Header,
-			SendTokenToUpstream: src.Spec.Override.AuthTypes.TestConsoleKey.SendTokenToUpstream,
+	if src.Spec.Default != nil {
+		defaultAuthv1Spec := src.Spec.Default
+		defaultAuthenticationSpec := v1alpha2.AuthSpec{}
+		defaultAuthenticationSpec.Disabled = defaultAuthv1Spec.Disabled
+		if defaultAuthv1Spec.AuthTypes != nil {
+			v1alpha2authTypes := v1alpha2.APIAuth{}
+			v1alpha2authTypes.Oauth2 = v1alpha2.Oauth2Auth{
+				Required:            "mandatory",
+				Disabled:            defaultAuthv1Spec.AuthTypes.Oauth2.Disabled,
+				Header:              defaultAuthv1Spec.AuthTypes.Oauth2.Header,
+				SendTokenToUpstream: defaultAuthv1Spec.AuthTypes.Oauth2.SendTokenToUpstream,
+			}
+			// Convert Oauth2Auth Default to v1alpha2.APIKey : Required field added as optional for APIKey
+			for _, apiKeyAuth := range defaultAuthv1Spec.AuthTypes.APIKey {
+				convertedAPIKeyAuth := v1alpha2.APIKeyAuth{
+					In:                  apiKeyAuth.In,
+					Name:                apiKeyAuth.Name,
+					SendTokenToUpstream: apiKeyAuth.SendTokenToUpstream,
+				}
+				v1alpha2authTypes.APIKey = append(v1alpha2authTypes.APIKey, convertedAPIKeyAuth)
+			}
+			if defaultAuthv1Spec.AuthTypes.TestConsoleKey != (TestConsoleKeyAuth{}) {
```

Comment:
does this != work when it empty struct is given? I'm not sure golang assign same address for empty object and != checks for address right?